### PR TITLE
Ra 12 order page(user side)

### PR DESCRIPTION
**Pull request by antonpby**

### CODE REVIEW

File: API/Controllers/OrderController.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+            if (orders == null) return NotFound();
+
+            return Ok(orders);
+        }
+
+        [HttpPost("new")]
+        public async Task<ActionResult> AddOrder(Order order)
+        {
+            order.OrderDate = DateTime.Now;
```
Comment (IKpassah) [inline API/Controllers/OrderController.cs:63]: i think we should get DateTime.UtcNow

File: API/Controllers/ProductController.cs [hunk 2]
```diff
@@ -50,10 +50,10 @@ public async Task<ActionResult<RealProductSchedule>> GetRealProductShedule(int i
         {
             //would test it later
             return await _dataContext.RealProducts
+                .Where(p => p.Id == id)
                 .Include(rp => rp.OrderProducts).ThenInclude(op => op.Order)
                 .ProjectTo<RealProductSchedule>(_mapper.ConfigurationProvider).FirstOrDefaultAsync();
```
Comment (IKpassah) [inline API/Controllers/ProductController.cs:55]: RealProductScheduleDto ?

File: client/src/app/_cart/card-item/card-item.component.ts [hunk 3]
```diff
@@ -9,22 +15,114 @@ import { SelectProductService } from 'src/app/_services/select-product.service';
...
+      realProductId: this.item.id
+    }
+    orderProducts.push(orderProduct);
+
+    let newOrder: CreateOrder = {
+      orderProducts : orderProducts,
+      requiredDate : this.myForm?.controls.requiredDate.value,
+      requiredReturnDate : this.myForm?.controls.requiredReturnDate.value,
+      comments : this.myForm?.controls.comments?.value,
+      customeId : user.id,
```
Comment (IKpassah) [inline client/src/app/_cart/card-item/card-item.component.ts:66]: customerId

File: client/src/app/_cart/card-item/card-item.component.ts [hunk 4]
```diff
@@ -9,22 +15,114 @@ import { SelectProductService } from 'src/app/_services/select-product.service';
...
-    this.selectProductService.remove(this.item);
-    this.show = false;
+  addOrder(){
+    
+    let user = JSON.parse(localStorage.getItem('user'));
+    let orderProducts: OrderProduct[] = [];
+    let orderProduct: OrderProduct = {
+      realProductId: this.item.id
+    }
+    orderProducts.push(orderProduct);
```
Comment (IKpassah) [inline client/src/app/_cart/card-item/card-item.component.ts:59]: if server didn't add order. user will see product as added whatever. I think in line 71 you sould subscribe. and after successfull response add order to list

File: client/src/app/_models/createOrder.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { OrderProduct } from "./orderProduct";
+
+    export interface CreateOrder {
+        orderProducts: OrderProduct[];
+        requiredDate: Date;
+        requiredReturnDate: Date;
+        comments: string;
+        customeId: number;
```
Comment (IKpassah) [inline client/src/app/_models/createOrder.ts:8]: customerId

File: client/src/app/_models/order.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    id: number;
+    orderProducts: OrderProduct[];
+    orderDate: Date;
+    requiredDate: Date;
+    shippedDate: Date;
+    requiredReturnDate: Date;
+    returnDate: Date;
+    status: string;
+    comments: string;
+    customeId: number;
```
Comment (IKpassah) [inline client/src/app/_models/order.ts:13]: also

File: client/src/app/_services/order.service.ts [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+  getOrdersByUserId(userId: number){
+    return this.http.get<OrderDto[]>(this.baseUrl + 'order/list/' + userId)
+  }
+
+
+  addOrder(order: CreateOrder){
+    console.log(JSON.stringify(order));
+    let q = 1;
+    return this.http.post(this.baseUrl + 'order/new', order).subscribe(q => {
```
Comment (IKpassah) [inline client/src/app/_services/order.service.ts:26]: subscribe should  be out of service

**Review by IKpassah — CHANGES_REQUESTED** [review]
i put some comments. plz change it

**Review by NicoBSU — CHANGES_REQUESTED** [review]
See all change requests of Vanya

**Pull request merged**